src/main/java/org/broadinstitute/dsm/model/dynamicdashboard/graph/heatmap/KitStatusesHeatMap.java

```diff
@@ -0,0 +1,245 @@
+
+        Map<String, String> participantsGuidsWithShortIds = getParticipantsGuidsWithShortIds(ddpParticipantsFromES);
+
+        Map<String, List<DDPKitRequestDto>> participantsWithKits = getParticipantsWithKitsCreatedBySystem(participantsGuidsWithShortIds);
+
+        Map<String, Map<String, List<Object>>> participantsWithKitsSeparatedByMonth =
+                getParticipantsWithKitsSeparatedByMonth(ddpParticipantsFromES, participantsWithKits);
+
+        List<HeatmapGraphData.HeatMapRow> heatmapGraphRows = getHeatmapRows(ddpParticipantsFromES, participantsWithKitsSeparatedByMonth);
+
+        List<HeatmapGraphData.HeatMapDataSet> heatMapDataSet = getHeatmapDataset(participantsWithKitsSeparatedByMonth);
+
+        List<Map<String, Object>> columns = getColumns();
+        Map<String, Object> colorRange = getColorRange();
+        return new HeatmapGraphData(heatmapGraphRows, columns, colorRange, heatMapDataSet, getStatisticPayload().getDisplayType(), getStatisticPayload().getDisplayText());
+    }
+
+    List<HeatmapGraphData.HeatMapDataSet> getHeatmapDataset(Map<String, Map<String, List<Object>>> participantsWithKitsSeparatedByMonth) {
+        List<HeatmapGraphData.HeatMapDataSet> heatMapDataSet = new ArrayList<>();
+        participantsWithKitsSeparatedByMonth.forEach(
+                (pId, value) -> value.forEach((k, v) -> heatMapDataSet.add(new HeatmapGraphData.HeatMapDataSet(pId, k, getColorRange(v)))));
+        return heatMapDataSet;
+    }
+
+    List<HeatmapGraphData.HeatMapRow> getHeatmapRows(Map<String, Map<String, Object>> ddpParticipantsFromES,
+                                                         Map<String, Map<String, List<Object>>> participantsWithKitsSeparatedByMonth) {
+        return participantsWithKitsSeparatedByMonth.keySet().stream()
+                .map(shortId -> new HeatmapGraphData.HeatMapRow(shortId, shortId, Participant
+                        .getParticipantGuidByHruidFromParticipantESData(ddpParticipantsFromES, shortId)))
+                .collect(Collectors.toList());
+    }
+
+    Map<String, String> getParticipantsGuidsWithShortIds(Map<String, Map<String, Object>> ddpParticipantsFromES) {
+        Map<String, String> participantsGuidsWithShortIds = new HashMap<>();
+        ddpParticipantsFromES.forEach((k, v) -> {
+            String shortId = (String)((Map) v.get(ElasticSearchUtil.PROFILE)).get(ElasticSearchUtil.HRUID);
+            participantsGuidsWithShortIds.put(k, shortId);
+        });
+        return participantsGuidsWithShortIds;
+    }
+
+    private Map<String, List<DDPKitRequestDto>> getParticipantsWithKitsCreatedBySystem(Map<String, String> ddpParticipantsFromES) {
+        Map<String, List<DDPKitRequestDto>> participantsWithKits = new HashMap<>();
+        DDPKitRequestDao ddpKitRequestDao = new DDPKitRequestDao();
+        ddpParticipantsFromES.forEach((pId, shortId) -> participantsWithKits.put(shortId,
+                ddpKitRequestDao.getKitRequestsByParticipantId(pId).stream()
```

Comment:
can we refactor this: if I understand correctly, if we are displaying 1000 pts on the headmap we are querying DSM kit tables 1000 times for them?